crates/apub/src/api/user_settings_backup.rs

```diff
@@ -133,7 +133,8 @@ pub async fn import_settings(
     local_user_view.local_user.id,
     &local_user_form,
   )
-  .await?;
+  .await
```

Comment:
Rather than doing this, check this out from the diesel update docs:

> It should be noted that receiving 0 rows from [get_result](https://docs.diesel.rs/2.1.x/diesel/query_dsl/trait.RunQueryDsl.html#method.get_result) is considered an error condition by default. If you want to get back 0 or 1 row (e.g. have a return type of QueryResult<Option<T>>), then you will need to call .get_result(...).optional().

So I'd recommend doing the same thing I did for inserts, but for all updates: adding `.optional()` so that we can catch the runtime errors. IMO this should be done for all updates, so we can handle that case.

If you want me to handle this, lmk.

Reply:
That would not hadle the case of all fields in the update form being set to None, which throws QueryBuilderError

Reply:
I'm going to open up an issue in diesel for this, bc it needs a similar solution to `optional()` then, or optional should also handle that case.

Reply:
In the meantime, you can do a match on that specific `err.kind()` https://stackoverflow.com/a/43884600/1655478

Reply:
In this case, it should be `Err(Error::QueryBuilderError(_)) =>`